src/web/mjs/connectors/MangaDex.mjs

```diff
@@ -47,66 +47,59 @@ export default class MangaDex extends Connector {
-        const {data} = await this.fetchJSON(request);
+        const { data } = await this.fetchJSON(request);
         return new Manga(this, id, data.attributes.title.en || Object.values(data.attributes.title).shift());
     }
 
     async _getMangas() {
         let mangaList = [];
-        // NOTE: Due to a limit of 10k results, it is necessary to permutate some filters in order to get as many results as possible
-        const queries = [
-            // [none] => total: 19000 (@2021/07)
-            { publicationDemographic: 'none', order: 'asc' },
-            { publicationDemographic: 'none', order: 'desc' },
-            // [josei] => total: 2396 (@2021/07)
-            { publicationDemographic: 'josei', order: 'asc' },
-            // [seinen] => total: 6810 (@2021/07)
-            { publicationDemographic: 'seinen', order: 'asc' },
-            // [shoujo] => total: 7441 (@2021/07)
-            { publicationDemographic: 'shoujo', order: 'asc' },
-            // [shounen] => total: 7436 (@2021/07)
-            { publicationDemographic: 'shounen', order: 'asc' },
-        ];
-        for(let query of queries) {
-            for(let page = 0, run = true; run; page++) {
-                let mangas = await this._getMangasFromPage(page, query);
-                mangas.length > 0 ? mangaList.push(...mangas) : run = false;
-            }
-        }
-        return mangaList;
-    }
+        let data1 = await this._getMangasFromPages(0, 99);
+        mangaList = [...mangaList, ...data1.data];
+        let nextAt = data1.nextAt;
 
-    async _getMangasFromPage(page, query) {
-        if(page > 99) {
-            // NOTE: Current limit is 10000 entries (elastic search limit)
-            //       => Do not throw an error but return gracefully instead, so at least the partial list is shown
-            return [];
+        for (let i = 1; i <= data1.total / 10000; i += 1) {
+            let data2 = await this._getMangasFromPages(0, 0, nextAt);
+            mangaList = [...mangaList, ...data2.data.slice(1)];
+            let pages = Math.min(Math.floor(data2.total / 100), 99);
+            if (pages > 0) {
+                let data3 = await this._getMangasFromPages(1, pages, nextAt);
+                mangaList = [...mangaList, ...data3.data];
+                nextAt = data3.nextAt;
+            }
         }
-        await this.wait(this.throttleGlobal);
-        const uri = new URL('/manga', this.api);
-        uri.searchParams.set('limit', 100);
-        uri.searchParams.set('offset', 100 * page);
-        uri.searchParams.set('publicationDemographic[]', query.publicationDemographic);
-        uri.searchParams.set('order[createdAt]', query.order);
-        uri.searchParams.append('contentRating[]', 'safe');
-        uri.searchParams.append('contentRating[]', 'suggestive');
-        uri.searchParams.append('contentRating[]', 'erotica');
-        uri.searchParams.append('contentRating[]', 'pornographic');
```

Comment:
Removing this content rating will lead to missing more then 9000 mangas in the list
https://api.mangadex.org/manga?contentRating%5B%5D=safe&contentRating%5B%5D=suggestive&contentRating%5B%5D=erotica&contentRating%5B%5D=pornographic